core/lib/types/src/storage_writes_deduplicator.rs

```diff
@@ -107,26 +110,31 @@ impl StorageWritesDeduplicator {
                 &mut self.metrics.repeated_storage_writes
             };
 
+            let total_size = &mut self.metrics.total_updated_values_size;
```

Comment:
Maybe a stupid question: Could `metrics` be computed on demand based on `self.modified_key_values` instead of being contained in `StorageWritesDeduplicator`? IMO, if it's possible (and doesn't have significant computational overhead), it would be preferable because such computation would be stateless (doesn't need to take rollbacks into account etc.).